test/powershell/Modules/Microsoft.PowerShell.Utility/Get-Variable.Tests.ps1

```diff
@@ -90,7 +90,15 @@ Describe "Get-Variable" -Tags "CI" {
 		Get-Variable -Name var1 -ValueOnly | Should -Be 4
 
 		Remove-Variable var1
-    }
+	}
+
+	It 'Should return the same value with -ValueOnly as directly calling the variable returns' {
```

Comment:
This test will be very confusing for someone if it fails.  I think that this should do an explicit check to ensure a Get-Variable -ValueOnly collection is enumerated to the pipe, and not depend on Pester 'Should'.  Maybe something similar to the original repro where the two cases are piped to Get-Member, then check type name.

Also, I think the 'It' description should be made more clear, or at least add a comment providing a clearer explanation of the test.  Something about -ValueOnly should enumerate collections to output pipe.

Reply:
Get-Member is a bit of a chunky way to go, generating a lot of data we don't need... we ought to be able to do `$v | % GetType` and `gv v | % GetType` and compare instead with the same result.

I'll try to clarify the `It` name. 😄

Reply:
Okay, I _think_ that ought to be clear enough... let me know if I need to rethink it. 😄